src/merge_tools.rs

```diff
@@ -233,32 +229,27 @@ pub fn run_mergetool(
         .try_collect()?;
 
     let args = interpolate_mergetool_filename_patterns(&editor.merge_args, &paths);
-    let args_str = args
-        .iter()
-        .map(|p| p.to_string_lossy().into_owned())
-        .collect_vec();
-    let exit_status = Command::new(&editor.program)
-        .args(args)
-        .status()
-        .map_err(|e| ExternalToolError::FailedToExecute {
-            tool_binary: editor.program.clone(),
-            args: args_str.clone(),
-            source: e,
-        })?;
+    let mut cmd = Command::new(&editor.program);
+    let exit_status = cmd.args(args).status();
+    // TODO: Have a --debug flag for debugging information (see other uses of
+    //    tracing::debug) a separate --verbose flag for printing this kind of
+    //    information.
+    // TODO: Remove the "(run with --verbose to see the exact invocation)"
+    //    from the errors. Print it as a hint but only if --verbose is *not* set.
+    tracing::debug!(?cmd, "Attempted to run an external merge tool: ");
+    let exit_status = exit_status.map_err(|e| ExternalToolError::FailedToExecute {
```

Comment:
Maybe better to move tracing output before `.status()`. Merge tool may print its information, and we would want to see tracing events in that order.

Reply:
No idea why. It works for me. (Tested with `edit_diff()`)
```
    let mut cmd = Command::new(&editor.program);
    cmd
        .args(&editor.edit_args)
        .arg(&left_wc_dir)
        .arg(&right_wc_dir);
    tracing::debug!(?cmd);
    let exit_status = cmd.status()
```

Reply:
You are right, this works. Not sure how it differs from what I did. Thanks!